### PR TITLE
Fix D3D8 Issues

### LINKED ISSUE #64: Errors with current OOVPA database
The errors were found after call XbSymbolDataBaseTestOOVPAs function. Other symbol libraries had not report any error. This issue is a reminder of current issues in the codebase.

```
ERROR  : OOVPATable  1 4531 [  27] D3DDevice_BeginPush, comparing against OOVPATable  2 4039 [ 412] D3DDevice_BeginPush2 : OOVPA's are expanded (left +2, right +2)
ERROR  : OOVPATable  1 4039 [  30] D3DDevice_BeginPush2, comparing against OOVPATable  2 4531 [ 409] D3DDevice_BeginPush : OOVPA's are expanded (left +2, right +2)
ERROR  : OOVPATable  1 5344 [ 101] D3DDevice_DrawIndexedVerticesUP, comparing against OOVPATable  2 1024 [  53] D3DDevice_DrawVerticesUP : OOVPA's are expanded (left +4, right +4)
ERROR  : OOVPATable  1 4831 [ 239] D3DDevice_PersistDisplay, comparing against OOVPATable  2 5558 [ 623] D3DDevice_PersistDisplay : OOVPA's are expanded (left +4, right +3)
ERROR  : OOVPATable  1 5558 [ 241] D3DDevice_PersistDisplay, comparing against OOVPATable  2 4831 [ 621] D3DDevice_PersistDisplay : OOVPA's are expanded (left +3, right +4)
ERROR  : OOVPATable  1 3911 [ 299] D3DDevice_SetRenderState_DoNotCullUncompressed, comparing against OOVPATable  2 1024 [ 153] D3DDevice_SetRenderState_DoNotCullUncompressed : OOVPA's are identical
ERROR  : OOVPATable  1 4034 [ 301] D3DDevice_SetRenderState_Dxt1NoiseEnable, comparing against OOVPATable  2 1036 [ 155] D3DDevice_SetRenderState_Dxt1NoiseEnable : OOVPA's are expanded (left +5, right +1)
ERROR  : OOVPATable  1 4034 [ 330] D3DDevice_SetRenderState_OcclusionCullEnable, comparing against OOVPATable  2 4034 [ 723] D3DDevice_SetRenderState_StencilCullEnable : OOVPA's are expanded (left +0, right +1)
ERROR  : OOVPATable  1 3911 [ 333] D3DDevice_SetRenderState_RopZCmpAlwaysRead, comparing against OOVPATable  2 1024 [ 193] D3DDevice_SetRenderState_RopZCmpAlwaysRead : OOVPA's are identical
ERROR  : OOVPATable  1 3911 [ 334] D3DDevice_SetRenderState_RopZRead, comparing against OOVPATable  2 1024 [ 194] D3DDevice_SetRenderState_RopZRead : OOVPA's are identical
ERROR  : OOVPATable  1 3911 [ 338] D3DDevice_SetRenderState_Simple, comparing against OOVPATable  2 4034 [ 721] D3DDevice_SetRenderState_Simple : OOVPA's are expanded (left +3, right +2)
ERROR  : OOVPATable  1 4034 [ 339] D3DDevice_SetRenderState_Simple, comparing against OOVPATable  2 3911 [ 720] D3DDevice_SetRenderState_Simple : OOVPA's are expanded (left +2, right +3)
ERROR  : OOVPATable  1 4034 [ 341] D3DDevice_SetRenderState_StencilCullEnable, comparing against OOVPATable  2 4034 [ 712] D3DDevice_SetRenderState_OcclusionCullEnable : OOVPA's are expanded (left +1, right +0)
ERROR  : OOVPATable  1 4242 [ 391] D3DDevice_SetSwapCallback, comparing against OOVPATable  2 4134 [ 837] D3DDevice_SetVerticalBlankCallback : OOVPA's are identical
ERROR  : OOVPATable  1 4134 [ 455] D3DDevice_SetVerticalBlankCallback, comparing against OOVPATable  2 4242 [ 773] D3DDevice_SetSwapCallback : OOVPA's are identical
ERROR  : OOVPATable  1 4034 [ 496] D3DVertexBuffer_Lock, comparing against OOVPATable  2 4531 [ 879] D3DVertexBuffer_Lock : OOVPA's are expanded (left +2, right +2)
ERROR  : OOVPATable  1 4531 [ 497] D3DVertexBuffer_Lock, comparing against OOVPATable  2 4034 [ 878] D3DVertexBuffer_Lock : OOVPA's are expanded (left +2, right +2)
ERROR  : OOVPATable  1 3911 [ 501] D3D_AllocContiguousMemory, comparing against OOVPATable  2 1060 [ 236] D3DDevice_SetRenderTarget : OOVPA's are expanded (left +4, right +5)
ERROR  : OOVPATable  2 1024 [  53] D3DDevice_DrawVerticesUP, comparing against OOVPATable  1 5344 [ 101] D3DDevice_DrawIndexedVerticesUP : OOVPA's are expanded (left +4, right +4)
ERROR  : OOVPATable  2 1024 [ 153] D3DDevice_SetRenderState_DoNotCullUncompressed, comparing against OOVPATable  1 3911 [ 299] D3DDevice_SetRenderState_DoNotCullUncompressed : OOVPA's are identical
ERROR  : OOVPATable  2 1036 [ 155] D3DDevice_SetRenderState_Dxt1NoiseEnable, comparing against OOVPATable  1 4034 [ 301] D3DDevice_SetRenderState_Dxt1NoiseEnable : OOVPA's are expanded (left +1, right +5)
ERROR  : OOVPATable  2 1024 [ 193] D3DDevice_SetRenderState_RopZCmpAlwaysRead, comparing against OOVPATable  1 3911 [ 333] D3DDevice_SetRenderState_RopZCmpAlwaysRead : OOVPA's are identical
ERROR  : OOVPATable  2 1024 [ 194] D3DDevice_SetRenderState_RopZRead, comparing against OOVPATable  1 3911 [ 334] D3DDevice_SetRenderState_RopZRead : OOVPA's are identical
ERROR  : OOVPATable  2 1060 [ 236] D3DDevice_SetRenderTarget, comparing against OOVPATable  1 3911 [ 501] D3D_AllocContiguousMemory : OOVPA's are expanded (left +5, right +4)
ERROR  : OOVPATable  2 4531 [ 409] D3DDevice_BeginPush, comparing against OOVPATable  1 4039 [  30] D3DDevice_BeginPush2 : OOVPA's are expanded (left +2, right +2)
ERROR  : OOVPATable  2 4039 [ 412] D3DDevice_BeginPush2, comparing against OOVPATable  1 4531 [  27] D3DDevice_BeginPush : OOVPA's are expanded (left +2, right +2)
ERROR  : OOVPATable  2 4831 [ 621] D3DDevice_PersistDisplay, comparing against OOVPATable  1 5558 [ 241] D3DDevice_PersistDisplay : OOVPA's are expanded (left +4, right +3)
ERROR  : OOVPATable  2 5558 [ 623] D3DDevice_PersistDisplay, comparing against OOVPATable  1 4831 [ 239] D3DDevice_PersistDisplay : OOVPA's are expanded (left +3, right +4)
ERROR  : OOVPATable  2 4034 [ 712] D3DDevice_SetRenderState_OcclusionCullEnable, comparing against OOVPATable  1 4034 [ 341] D3DDevice_SetRenderState_StencilCullEnable : OOVPA's are expanded (left +0, right +1)
ERROR  : OOVPATable  2 3911 [ 720] D3DDevice_SetRenderState_Simple, comparing against OOVPATable  1 4034 [ 339] D3DDevice_SetRenderState_Simple : OOVPA's are expanded (left +3, right +2)
ERROR  : OOVPATable  2 4034 [ 721] D3DDevice_SetRenderState_Simple, comparing against OOVPATable  1 3911 [ 338] D3DDevice_SetRenderState_Simple : OOVPA's are expanded (left +2, right +3)
ERROR  : OOVPATable  2 4034 [ 723] D3DDevice_SetRenderState_StencilCullEnable, comparing against OOVPATable  1 4034 [ 330] D3DDevice_SetRenderState_OcclusionCullEnable : OOVPA's are expanded (left +1, right +0)
ERROR  : OOVPATable  2 4242 [ 773] D3DDevice_SetSwapCallback, comparing against OOVPATable  1 4134 [ 455] D3DDevice_SetVerticalBlankCallback : OOVPA's are identical
ERROR  : OOVPATable  2 4134 [ 837] D3DDevice_SetVerticalBlankCallback, comparing against OOVPATable  1 4242 [ 391] D3DDevice_SetSwapCallback : OOVPA's are identical
ERROR  : OOVPATable  2 4034 [ 878] D3DVertexBuffer_Lock, comparing against OOVPATable  1 4531 [ 497] D3DVertexBuffer_Lock : OOVPA's are expanded (left +2, right +2)
ERROR  : OOVPATable  2 4531 [ 879] D3DVertexBuffer_Lock, comparing against OOVPATable  1 4034 [ 496] D3DVertexBuffer_Lock : OOVPA's are expanded (left +2, right +2)
```

EDIT: Fixed the error output to show the right build version.

TODO list:
- [ ] Some D3D offset need a manual work done than create many OOVPAs.
  * D3DDevice_SetSwapCallback (4242) and D3DDevice_SetVerticalBlankCallback (4134) are report as identical
  * And require a separate pull request to make the improvement.
- [ ] Something about D3DDevice_BeginPush (4531) and D3DDevice_BeginPush2 (4039) needs improvement + reference to known symbols.
- [ ] Resolve remaining D3D8 & D3D8LTCG "expanded" issues.

### PR DESCRIPTION
**Pull request by RadWolfie**
Resolve #64 (Errors with current OOVPA database)

Resolve the issue above plus an old bug discovered using segaboot did not trigger manual D3D scan as a requirement.

I had tested titles with D3D8:
- 3925
- 3948
- 4034
- 4039
- 4242
- 4531
- 4627
- 5028
- 5120
- 5455
- 5558
- 5788
- 5849

Plus D3D8LTCG:
- 3925
- 5344
- 5455
- 5659
- 5849

They had not made any regression since no symbols were lost. However, I need someone to test with LTCG titles.

### CODE REVIEW

**Comment by RadWolfie** [comment]
@jarupxx Could you verify with your LTCG titles?

**Comment by RadWolfie** [comment]
One thing to keep in mind, `D3D_MakeRequestedSpace` for LTCG may not have a rename. This can be left alone until further investigation to reduce the OOVPA signatures if possible.

**Comment by jarupxx** [comment]
@RadWolfie I tested with LTCG titles and did not regress.

- 3925
- 4039
- 4432
- 4531
- 4627
- 4721
- 4831
- 4928
- 5028
- 5120
- 5233
- 5344
- 5455
- 5558
- 5659
- 5788
- 5849

I also verified about D3D_MakeRequestedSpace.
`D3D_MakeRequestedSpace_1024` was supposed to have many LTCG titles, but offset 0x01 is incorrect (0x56 is correct, not 0x53). Use D3D8 symbol instead.
`D3D_MakeRequestedSpace_1036` has Manhunt title that Version 5849. You're rename D3D_MakeRequestedSpace_8 if you wish.

**Comment by RadWolfie** [comment]
Thanks for the verify, can you approve and merge the pull request?

File: OOVPADatabase/D3D8LTCG_OOVPA.inl [hunk 1]
```diff
@@ -65,7 +65,7 @@
 // ******************************************************************
 OOVPATable D3D8LTCG_OOVPAV2[] = {
 
-    REGISTER_OOVPAS(D3D_MakeRequestedSpace, 1024, 1036), //TODO: Need to verify, both versions are in 5849. Plus D3D8 symbol is detected.
+    REGISTER_OOVPAS(D3D_MakeRequestedSpace_8, 1036), // NOTE: The verison is in 5849. Being used by (5849) Manhunt title
```
Comment (JayFoxRox) [inline OOVPADatabase/D3D8LTCG_OOVPA.inl]: - "verison"?
- "is in 5849"?

The whole sentence is messed up (both of them?).

Comment (RadWolfie) [reply OOVPADatabase/D3D8LTCG_OOVPA.inl]: Rephased to `OOVPA is in 5849`, I was distracted when I made the change to match @jarupxx's comment.

**Pull request merged**